### PR TITLE
Middlewares, Routes handling and Response Parsing

### PR DESCRIPTION
**Pull request by sudharshanreddyam**

### CODE REVIEW

File: src/http_servers/html-server.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      res.writeHead(200, {'Content-Type': 'text/html'});
+      createReadableStream().pipe(res);
+    })
+  .listen(3002);  
+
+
+  function createReadableStream() {
+    const readStream = new Readable({
+      read() {
+        const htmlTemplate = readFileSync(join(__dirname, TEMPLATE_PATH), 'utf8');
```
Comment (AliakseiBychyk) [inline src/http_servers/html-server.js]: `fs` has its method `fs.createReadStream()`, the idea of hometask was to implement both approaches to send html content - with `fs.readFileSync()` and with `fs.createReadStream()` 
`fs.createReadStream()` will return the stream object, then you can pipe replace method to change content, and at the end pipe it to a response stream.

**Review by AliakseiBychyk — CHANGES_REQUESTED** [review]
Please pay attention to the p.4 of the hometask where the project structure is defined.

File: src/http_servers/html-server.js (hunk 1 above)
Comment (sudharshanreddyam) [reply src/http_servers/html-server.js]: My bad, I understood it wrong. I will update it accordingly.

Comment (sudharshanreddyam) [reply src/http_servers/html-server.js]: Updated the code to implement two approaches `fs.readFileSync` and `fs.createReadStream`

File: src/http_servers/html-server.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+.listen(3002);
+
+http.createServer((req, res) => {
+  res.writeHead(200, { 'Content-Type': 'text/html' });
+  const chunks = [];
+  const htmlStream = fs.createReadStream(path.join(__dirname, TEMPLATE_PATH));
+  htmlStream.on('data', (chunk) => {
+    chunks.push(chunk);
+  });
+  htmlStream.on('end', () => {
```
Comment (AliakseiBychyk) [inline src/http_servers/html-server.js]: Good. That approach works good. 
Just in order to be aware how to do it with pipes:
`http.createServer((req, res) => {`
    `  res.writeHead(200, { 'Content-Type': 'text/html' });`
    `  fs.createReadStream(path.join(__dirname, TEMPLATE_PATH))`
      `    .pipe(replace('{message}', 'any new html content'))`
      `    .pipe(res);`
  `}).listen(3009);`
the only thing to add is 'stream-replace' npm package, that way we can leverage stream's pipeline opportunities

**Review by AliakseiBychyk — APPROVED** [review]

File: src/http_servers/html-server.js (hunk 2 above)
Comment (sudharshanreddyam) [reply src/http_servers/html-server.js]: Thanks for the approval @AliakseiBychyk. I have tried to use pipes but didn't get any clue on how to replace the string. Thanks for the suggestion I would make a note of it.

Comment (AliakseiBychyk) [reply src/http_servers/html-server.js]: You should install 'stream-replace' npm package and don't forget to add
`import replace from 'stream-replace'` to the code

**Review by AliakseiBychyk — APPROVED** [review]

**Pull request merged**